### DISCUSSION TITLE
How to convert the `index` of `get_quote` and family to `Date`?

### DISCUSSION THREAD

**Discussion by andreltr** [serkor1/cryptoQuotes#17] [Q&A] [answered]
Hello,

From my understanding, the package output is a table such as the one below,
```
                        open     high     low    close    volume
2024-06-13 21:00:00 66773.01 67370.24 65078.0 66043.99 28408.188
2024-06-14 21:00:00 66043.99 66478.48 65857.1 66228.25 11451.802
```

Thus its first column is assumed to be its index.  Is this correct? Perhaps the return is a data frame?
Anyway, I converted it to a data frame.  Its time output isn´t important for me since I work on the day timestamp.
My question/doubt is if there would be some way to get only the date without its timestamp?

Best.

**Comment by serkor1** [comment, marked as answer]
Hi @andreltr,

Thank you for the question. Below is a *MWE* on how convert `POSIXct` to `Date`,

### Without conversion

``` r
# load BTC in
# daily intervals and 
# display results
tail(
  BTC <- cryptoQuotes::get_quote(
    ticker   = "PF_XBTUSD",
    source   = "kraken",
    interval = "1d"
  )
)
#>                      open  high   low close    volume
#> 2024-06-15 02:00:00 66015 66462 65825 66199  1743.264
#> 2024-06-16 02:00:00 66199 66957 66003 66660  1361.403
#> 2024-06-17 02:00:00 66660 67290 65075 66485  7297.915
#> 2024-06-18 02:00:00 66485 66573 64028 65150 10071.968
#> 2024-06-19 02:00:00 65150 65710 64620 64956  5108.370
#> 2024-06-20 02:00:00 64956 65640 64825 65440  1317.906
```

### Converting to `Date`

One option is to modify the `index` directly via `{zoo}` as show below,

``` r
# recoding the index to 
# date class
zoo::index(BTC) <- as.Date(
  zoo::index(BTC)
)

# display results
tail(
  BTC
)
#>             open  high   low close    volume
#> 2024-06-15 66015 66462 65825 66199  1743.264
#> 2024-06-16 66199 66957 66003 66660  1361.403
#> 2024-06-17 66660 67290 65075 66485  7297.915
#> 2024-06-18 66485 66573 64028 65150 10071.968
#> 2024-06-19 65150 65710 64620 64956  5108.370
#> 2024-06-20 64956 65640 64825 65440  1317.906
```

Then you can convert to `data.frame`, `tibble` or `data.table`. 

<sup>Created on 2024-06-20 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

--------

I would recommend looking into [{reprex}](https://github.com/tidyverse/reprex) for creating questions and issues.
